online-shop/src/Components/EditProductPage.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { SyntheticEvent } from 'react';
+import '../Styles/ComponentsStyles/ProductDetails.scss';
+import { AppState } from '../ReduxStore';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { IProduct } from '../Models/Models';
+import { initialProductDetailsLoad } from '../ReduxStore/EditProductPageSection/actions';
+import { Dispatch } from 'redux';
+import { LOAD_PRODUCT_DETAILS_FOR_EDIT, EDIT_PRODUCT } from '../ReduxStore/EditProductPageSection/types';
+
+interface EditProductPageProps {
+    productInEditStage: IProduct;
+    isLoading: boolean;
+    submitChangesResponse: number;
+    loadProductData: (productID: number) => void;
+    editProduct: (product: IProduct) => void;
+    match: any;
+    initialProductDetailsLoad: () => void;
+}
+
+interface AdditionalEditProductPageState {
+    match: any;
+}
+
+class EditProductPage extends React.Component<EditProductPageProps> {
+    productTitle: string = "";
+    productCategory: string = "";
+    productPrice: number = 0;
+    productImage: string = "";
+    productDescription: string = "";
+
+    onTitleChange = (e: SyntheticEvent) => {
+        this.productTitle = (e.target as HTMLInputElement).value.trim();
+    }
+
+    onCategoryChange = (e: SyntheticEvent) => {
+        this.productCategory = (e.target as HTMLInputElement).value.trim();
+    }
+
+    onPriceChange = (e: SyntheticEvent) => {
+        this.productPrice = +(e.target as HTMLInputElement).value.trim();  //string -> number conversion : +string = number
+    }
+
+    onImageChange = (e: SyntheticEvent) => {
+        this.productImage = (e.target as HTMLInputElement).value.trim();
+    }
+
+    onDescriptionChange = (e: SyntheticEvent) => {
+        this.productDescription = (e.target as HTMLInputElement).value.trim();
+    }
+
+    //String comparison function
+    areStringsEqual = (firstString: string, secondString: string): boolean => {
+        if (firstString === secondString) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    didFormInputValuesChanged = (): boolean => {
+        if (this.areStringsEqual(this.productTitle, this.props.productInEditStage.name) &&
+            this.areStringsEqual(this.productCategory, this.props.productInEditStage.category) &&
+            (this.productPrice === this.props.productInEditStage.price) &&
+            this.areStringsEqual(this.productImage, this.props.productInEditStage.image) &&
+            this.areStringsEqual(this.productDescription, this.props.productInEditStage.description)) {
+
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    areNewFormValuesOk = (title : string, image : string, description : string, price : string): boolean => {
+        if (title.length !== 0) {
```

Comment:
Pretty ugly method :) you could reverse the conditions and make the if-else sequential:

```
if (title.length === 0) {
   alert("Product title is missing");
} else if (image.length === 0) {
   alert("Product image is missing");
} // ...
```